cdap-app-templates/cdap-etl/cdap-etl-tools/src/main/java/co/cask/cdap/etl/tool/UpgradeTool.java

```diff
@@ -157,7 +179,10 @@ public static void main(String[] args) throws Exception {
         "as expected by older versions of the etl artifacts."))
       .addOption(new Option("o", "outputfile", true, "File to write the converted application details provided in " +
         "the configfile option. If none is given, results will be written to the input file + '.converted'. " +
-        "The contents of this file can be sent directly to CDAP to update or create an application."));
+        "The contents of this file can be sent directly to CDAP to update or create an application."))
+      .addOption(new Option("e", "errorDir", true, "Optional directory to write any upgraded pipeline configs that " +
+        "failed to upgrade. The problematic configs can then be manually edited and upgraded separately. " +
+        "Upgrade errors may happen for pipelines that use plugins that are not backwards compatible."));
```

Comment:
it has to be writable by whoever is running the tool.  I can add that to the description.  Can also add some validation on the directory before upgrade starts.